scripts/generate/templates/vanilla.template.ts

```diff
@@ -49,15 +49,21 @@ async function getRenderData() {
  * optionally make use of data stored during `ShouldRender` extension point to
  * expedite time-to-first-meaningful-paint.
  */
-extend('Checkout::PostPurchase::Render', (root, input) => {
-  const payload = input.storage.initialData as Payload;
+extend('Checkout::PostPurchase::Render', (root, {extensionPoint, storage}) => {
+  const initialState = storage.initialData as InitialState;
   const text = root.createComponent(Text);
   text.appendChild(
-    `Create your awesome offer page here.  ShouldRender payload=${JSON.stringify(
-      payload
+    `Create your awesome post purchase page here. initialState=${JSON.stringify(
+      initialState
     )}`
   );
+  const button = root.createComponent(Button);
+  button.props.onPress = () => {
```

Comment:
Does this actually work? `props` are frozen so I was hoping no, and it definitely doesn't have any special setters in place to actually apply this change. Since you are constructing the button directly above, you can pass the props as the second argument instead.

Reply:
It does work; I will make the change you suggest.